notebooks/faq.clj

```diff
@@ -0,0 +1,62 @@
+
+**Why Wolframite?**
+
+'Why Wolframite?' is really just the combination of 'Why Wolfram?' (above) and '[Why Clojure?](https://clojure.org/about/rationale)'.
+
+In a nutshell, the Wolfram programming language is 'best-in-class' for certain specialist tasks, like manipulating equations, but, to paraphrase (Randall Munroe)[https://xkcd.com/1270/],
+> Wolfram combines the flexibility and power of abstract mathematics with the intuitive clarity of abstract mathematics.
+
+Clojure, on the other-hand, is a first-class, general-purpose, programming language whose core namespace and dynamical, functional, paradigm is well suited to data exploration and manipulation. So why don't we call Wolfram from Clojure? It seems like the atypical 'best of both worlds'.
+
+To put it another way, Wolfram is *a powerful tool in need of a toolbox*. And so what language is best placed to be that 'box'? Well, Wolfram is, underneath, built on LisP and so will naturally gel more with a similar design philosophy (and way of thinking) than, say, with an object-oriented or scripting language. Furthermore, somewhat ironically, it has a very good 'link' to the Java ecosystem. In [fact](https://reference.wolfram.com/language/JLink/tutorial/Introduction.html#25629),
+>  For all types of WSTP programs, J/Link provides a higher-level layer of functionality than the traditional C WSTP programming interface. This makes Java the easiest and most convenient language for writing programs to interact with the Wolfram Language.
+
+Even at a fundamental level therefore, our toolbox language should be a LisP with strong Java interop: mmm...!
+
+At the *usability* layer, Clojure is a well-designed, ergonomic language that leaves its users [happy](https://www.computerworld.com/article/1379902/clojure-developers-are-the-happiest-developers.html)! When it comes to working with mathematical expressions specifically, there are a few key features. First of all, for many people, Clojure was the first language to introduce ergonomic chaining of function expressions, *e.g.*
+```clojure
+(TeX->
+   (w/* 'x w/I)
+   (w/+ 'y)
+   ||2)
+```
+. In my view, chaining Wolfram function calls together with threading macros is actually a big usability improvement. Wolfram expressions can get pretty involved (it's common to end up with expressions that hold 10s of symbols and operators and 100s are not unheard of) and trying to read and manipulate these from the inside out is just not natural for the average human. It stands to reason then that chaining functions together (and debugging them!) can really be a pain. In fact, Wolfram recognised this problem when it introduced the prefix operator, `@`, to help with function composition, *e.g.* `f@g@h`. Unfortunately however, this doesn't work with multiple arguments. It is possible to do things like `f@@args`, and even things like `f@@@{{a, b}, {c, d}}`, but the readability quickly becomes dire. On the other hand, Clojure's threading is simple, clear and scalable.
```

Comment:
IMO, a scientist wouldn't necessarily know what a "threading macro" is. Perhaps clarify with ".. with threading macros..." -> "...with threading macros, as we do here, ..." and making `threading macros` into a link to a page that explains them?

Reply:
added 

[threading macros](https://clojure.org/guides/threading_macros) (*e.g.* `->` and `->>` etc.)